### PR TITLE
Por 19 home page second section

### PR DESCRIPTION
**Pull request by nicocamara**

### CODE REVIEW

**Review by nicocamara — COMMENTED** [review]
*

File: src/client/components/atoms/itemList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import getAssetUrl from '../../../utils/getAssetUrl';
+import Button from '../button';
+import './style.scss';
+
+type ItemListProps = {
+  item: Portfolio;
+};
+
+const ItemList = ({ item }: ItemListProps) => {
+  console.log('si', item);
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/index.tsx:11]: `no debugging code in PR`

, if prettier/eslint/ts complains about format, change it to a implicit return.

File: src/client/components/atoms/itemList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <div className="itemList">
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/index.tsx:13]: wrong use of BEM pattern - we don't use camelCase in classNames 

the block should be something like `item-list`

File: src/client/components/atoms/itemList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      {item.overview.avatarURL ? (
+        <img className="itemList__image " src={item.overview.avatarURL} />
+      ) : (
+        <img className="itemList__image " src={getAssetUrl('placeholder.png')} />
+      )}
+      <div className="itemList__dataContainer">
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/index.tsx:19]: wrong use of BEM pattern - we don't use camelCase in classNames

the element className should be something like `item-list__data-container` instead

File: src/client/components/atoms/itemList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        <div className="itemList__title">{item.overview.firstName + ' ' + item.overview.lastName}</div>
+
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/index.tsx:21]: no empty lines inside JSX

File: src/client/components/atoms/itemList/style.scss [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  height: 300px;
+  width: 240px;
+  padding: 10px;
+  margin: 5px;
+  background-color: white;
+
+  &:hover {
+    box-shadow: 8px 8px 5px rgb(154, 166, 170);
+  }
+  @media (max-width: $mobile) {
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/style.scss:19]: leave empty lines between CSS's selectors for better future comprehension

File: src/client/components/atoms/itemList/style.scss [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }
+  }
+
+  &__dataContainer {
+    display: flex;
+    flex-direction: column;
+    margin: 5px auto;
+    text-align: center;
+  }
+  &__title {
```
Comment (ignaciopetrone) [inline src/client/components/atoms/itemList/style.scss:55]: leave empty lines between CSS's selectors for better future comprehension

File: src/client/components/molecules/homeSections/second/index.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { useContext, useEffect, useState } from 'react';
+import ItemList from '../../../atoms/itemList';
+import './style.scss';
```
Comment (ignaciopetrone) [inline src/client/components/molecules/homeSections/second/index.tsx:3]: format and sort on every module you work.

`windows+shift+f` and `windows+shift+o` OR `alt+shift+f` and `alt+shift+o`(idk how is on windows)

File: src/client/components/molecules/homeSections/second/index.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import StateContext from '../../../../utils/stateContext';
+import { Portfolio } from '../../../../utils/Type';
+
+const SecondSection = () => {
+  const { handlers } = useContext(StateContext);
+  const [portfolioItems, setPortfolioItems] = useState<Portfolio[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      const response = await handlers.getAllPortfolios();
```
Comment (ignaciopetrone) [inline src/client/components/molecules/homeSections/second/index.tsx:13]: this request will get expensive in future. what is the pourpose of this?

if the idea is to show some existing portfolios as examples, better to use fake ones.

File: src/client/components/molecules/homeSections/second/index.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+      console.log('lola', response);
+
+      setPortfolioItems(response);
+    })();
+  }, []);
+
+  return (
+    <div className="secondSection">
```
Comment (ignaciopetrone) [inline src/client/components/molecules/homeSections/second/index.tsx:21]: wrong use of BEM pattern - we don't use camelCase in classNames

the block should be something like `secont-section` (although is better if the block's className is just one word)

File: src/client/components/molecules/homeSections/second/index.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+      <div className="secondSection__title">Check out some users who use our service</div>
+      <hr className="secondSection__line" />
+
+      <div className="secondSection__items">
+        {portfolioItems.map((item, index) => (
+          <ItemList key={index} item={item} />
```
Comment (ignaciopetrone) [inline src/client/components/molecules/homeSections/second/index.tsx:27]: when mapping, always try to set a key coming from the element in iteration

File: src/client/components/pages/Home/index.tsx [hunk 11]
```diff
@@ -6,11 +7,13 @@ const Home = () => (
...
         <div className="home__overview-text">Your professional portfolio whenever you need it</div>
+        <div className="home__section home__features">
+          <SecondSection />
+        </div>
         <div className="home__section home__features">
           <Slider />
         </div>
-        <div className="home__section home__features">commin</div>
-        <div className="home__section home__features">soon</div>
+            <div className="home__section home__features">soon</div>
```
Comment (ignaciopetrone) [inline src/client/components/pages/Home/index.tsx:16]: the three sections have the same 2 classes. I'd say the lastone `home__features` is incorrect or not necessary

File: src/client/utils/stateContext.tsx [hunk 12]
```diff
@@ -86,6 +87,13 @@ export const StateProvider = ({ children }: any) => {
     });
   };
 
+  //itentamos traer los portfolios
```
Comment (ignaciopetrone) [inline src/client/utils/stateContext.tsx:90]: `no comments on PR`

If the code needs a comment for better understanding, then the problem is the naming

File: src/client/utils/stateContext.tsx [hunk 13]
```diff
@@ -86,6 +87,13 @@ export const StateProvider = ({ children }: any) => {
     });
   };
 
+  //itentamos traer los portfolios
+  const getAllPortfolios = async () => {
+    const response = await callApi({ method: 'GET', endpoint: '/portfolios' });
+    console.log('response', response);
```
Comment (ignaciopetrone) [inline src/client/utils/stateContext.tsx:93]: no debugging code in PR

File: src/client/utils/stateContext.tsx [hunk 14]
```diff
@@ -86,6 +87,13 @@ export const StateProvider = ({ children }: any) => {
...
   };
 
+  //itentamos traer los portfolios
+  const getAllPortfolios = async () => {
+    const response = await callApi({ method: 'GET', endpoint: '/portfolios' });
+    console.log('response', response);
+    return response;
+  };
+
   // const getFile = async () => {
```
Comment (ignaciopetrone) [inline src/client/utils/stateContext.tsx:97]: I know you did not add this comments in the PR. but as general rule: 
`if is not in use, it should be removed`

File: src/server/utils/api.ts [hunk 15]
```diff
@@ -82,6 +82,24 @@ API.get('/portfolio/:userName', async (_req, res) => {
   }
 });
 
+//consulta de la Api
```
Comment (ignaciopetrone) [inline src/server/utils/api.ts:85]: no comments on PR

the endpoint is clear enough

File: src/server/utils/api.ts [hunk 16]
```diff
@@ -82,6 +82,24 @@ API.get('/portfolio/:userName', async (_req, res) => {
   }
 });
 
+//consulta de la Api
+API.get('/portfolios', async (_req, res) => {
```
Comment (ignaciopetrone) [inline src/server/utils/api.ts:86]: This endpoint is unnecessary, will get exponentially expensive while the project grows and more users registers

should be removed

File: src/server/utils/api.ts [hunk 17]
```diff
@@ -82,6 +82,24 @@ API.get('/portfolio/:userName', async (_req, res) => {
   }
 });
 
+//consulta de la Api
+API.get('/portfolios', async (_req, res) => {
+  try {
+    const portfolioRef = await db.collection('Portfolio').get();
+    const portfolios: Portfolio[] = [];
+
+    portfolioRef.forEach(doc => {
```
Comment (ignaciopetrone) [inline src/server/utils/api.ts:91]: we don't do a forEach for pushing elements into another array.

better to use `map`

suggestion: const portfolios = portfolioRef.map(doc => (doc.data()) as Portfolio)

File: src/server/utils/api.ts [hunk 18]
```diff
@@ -82,6 +82,24 @@ API.get('/portfolio/:userName', async (_req, res) => {
...
+API.get('/portfolios', async (_req, res) => {
+  try {
+    const portfolioRef = await db.collection('Portfolio').get();
+    const portfolios: Portfolio[] = [];
+
+    portfolioRef.forEach(doc => {
+      portfolios.push(doc.data() as Portfolio);
+    });
+
+    console.log('Los portfolios:', portfolios);
```
Comment (ignaciopetrone) [inline src/server/utils/api.ts:95]: no debugging code in PR

File: src/server/utils/api.ts [hunk 19]
```diff
@@ -82,6 +82,24 @@ API.get('/portfolio/:userName', async (_req, res) => {
...
+    res.status(200).send(portfolios);
+  } catch (err) {
+    console.log(err);
+    res.status(500).send();
+  }
+});
+
```
Comment (ignaciopetrone) [inline src/server/utils/api.ts:102]: leave empty line between endpoints for a better future comprehension

**Pull request merged**